### PR TITLE
Bluetooth: host: free the adv instance if start legacy advertise fail 

### PR DESCRIPTION
**Pull request by anchao**
Bluetooth: host: free the adv instance if start legacy advertise fail

the adv instance should be delete correctly if advertise fail,
Otherwise the instance will unable to allocate if start the legacy again

Signed-off-by: chao an <anchao@xiaomi.com>

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -7778,6 +7779,11 @@ int bt_le_adv_start_legacy(const struct bt_le_adv_param *param,
 			  param->options & BT_LE_ADV_OPT_USE_IDENTITY);
 
 	return 0;
+
+failed:
+	adv_delete_legacy();
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:7784]: Thank you for the fix, although I would suggest to do it a bit different.
In `bt_le_adv_start`:
```
	err = bt_le_adv_start_legacy(param, ad, ad_len, sd, sd_len);
	if (err) {
		adv_delete_legacy();
	}

	return err;
```

That is similar to how it is handled for extended advertise case.

**Review by joerchan — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/hci_core.c (hunk 1 above)
Comment (anchao) [reply subsys/bluetooth/host/hci_core.c]: Sorry for reply late, I moved the adv instance allocate out of the le_adv_start_legacy(), so as to reuse and unify the adv_new_legacy() and adv_delete_legacy()

File: subsys/bluetooth/host/hci_core.c [hunk 2]
```diff
@@ -8129,24 +8128,25 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
...
 
-		if (!adv) {
-			return -ENOMEM;
-		}
+	if (!adv) {
+		return -ENOMEM;
+	}
 
+	if (IS_ENABLED(CONFIG_BT_EXT_ADV) &&
+			BT_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:8139]: Please fix the indentation here, should be 1 tab + 4 spaces, not 3 tabs.

**Review by joerchan — CHANGES_REQUESTED** [review]
Looks good, just need an indentation fix.

File: subsys/bluetooth/host/hci_core.c (hunk 2 above)
Comment (anchao) [reply subsys/bluetooth/host/hci_core.c]: Got

**Review by joerchan — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**